stratum/hal/lib/common/p4_service.cc

```diff
@@ -83,8 +83,7 @@ ::util::Status P4Service::Setup(bool warmboot) {
 ::util::Status P4Service::Teardown() {
   {
     absl::WriterMutexLock l(&controller_lock_);
-    node_id_to_controllers_.clear();
-    connection_ids_.clear();
+    node_id_to_controller_manager_.clear();
```

Comment:
why are you not purging `num_controller_connections_` ?

Reply:
good point, should be reset as well.